service/src/main/java/org/whispersystems/textsecuregcm/auth/BaseAccountAuthenticator.java

```diff
@@ -63,14 +63,14 @@ public BaseAccountAuthenticator(AccountsManager accountsManager, Clock clock) {
       AuthorizationHeader authorizationHeader = AuthorizationHeader.fromUserAndPassword(basicCredentials.getUsername(), basicCredentials.getPassword());
       Optional<Account>   account             = accountsManager.get(authorizationHeader.getIdentifier());
 
-      if (!account.isPresent()) {
+      if (isNull(account)) {
```

Comment:
I don't think Objects.isNull checks whether or not Optional's are present or empty. I think this will just check if the Optional is null.

Reply:
It looks like this change was reverted in #127.